tests/sdk_tests/test_cli.py

```diff
@@ -35,3 +37,11 @@ def test_version(cli_runner):
     result = cli_runner.invoke(cli, ["--version"])
     assert result.exit_code == 0, result.stdout
     assert __version__ in result.stdout
+
+
+def test_bad_docker_executable_env_var(cli_runner):
+    os.environ["TESSERACT_DOCKER_EXECUTABLE"] = "not-a-docker"
+
+    result = cli_runner.invoke(cli, ["ps"])
+    assert result.exit_code == 1
+    assert "not found" in result.stderr
```

Comment:
Something is off here. The `result.stderr` is `''` in the tests, but it shouldn't be:

```console
❯ TESSERACT_DOCKER_EXECUTABLE="podman-hpc" uv run tesseract ps
 [-] Executable `podman-hpc` not found.
 [x] Aborting
```

What am I missing?

Reply:
I assume the envvar isn't taking effect because you already modified settings? So you either need to run the CLI command in a fresh subprocess, or modify the settings object directly.